File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/StoreTheirDidTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.isA;

import java.util.concurrent.ExecutionException;

public class StoreTheirDidTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testStoreTheirDidWorks() throws Exception {
		Signus.storeTheirDid(this.wallet, String.format("{\"did\":\"%s\"}", DID)).get();
	}

	@Test
	public void testCreateMyDidWorksForInvalidIdentityJson() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Signus.storeTheirDid(this.wallet, "{\"field\":\"value\"}").get();
	}

	@Test
	public void testStoreTheirDidWorksWithVerkey() throws Exception {
		Signus.storeTheirDid(this.wallet, String.format(IDENTITY_JSON_TEMPLATE, DID, VERKEY)).get();
	}

	@Test
	public void testStoreTheirDidWorksWithoutDid() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Signus.storeTheirDid(this.wallet, String.format("{\"verkey\":\"%s\"}", VERKEY)).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/DidTests/StoreTheirDidTest.cs
using Hyperledger.Indy.DidApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.DidTests
{
    [TestClass]
    public class StoreTheirDidTest : IndyIntegrationTestWithSingleWallet
    {
        private const string _verkey = "GjZWsBLgZCR18aL468JAT7w9CZRiBnpxUPPgyQxh4voa";
             
        [TestMethod]
        public async Task TestStoreTheirDidWorks()
        {
            await Did.StoreTheirDidAsync(wallet, string.Format("{{\"did\":\"{0}\"}}", DID));
        }

        [TestMethod]
        public async Task TestCreateMyDidWorksForInvalidIdentityJson()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                Did.StoreTheirDidAsync(wallet, "{\"field\":\"value\"}")
            );            
        }

        [TestMethod]
        public async Task TestStoreTheirDidWorksWithVerkey()
        {
            var json = string.Format(IDENTITY_JSON_TEMPLATE, DID, _verkey);

            await Did.StoreTheirDidAsync(wallet, json);
        }

        [TestMethod]
        public async Task TestStoreTheirDidWorksWithoutDid()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                Did.StoreTheirDidAsync(wallet, string.Format("{{\"verkey\":\"{0}\"}}", _verkey))
            );
        }
    }
}
